ext/opentelemetry-ext-otlpexporter/src/opentelemetry/ext/otlpexporter/trace_exporter/__init__.py

```diff
@@ -0,0 +1,305 @@
+                                    **translate_key_values(key, value)
+                                )
+                            )
+                        # pylint: disable=broad-except
+                        except Exception as error:
+                            logger.exception(error)
+
+                    collector_span_kwargs["events"].append(
+                        collector_span_event
+                    )
+
+            if sdk_span.links:
+                collector_span_kwargs["links"] = []
+
+                for sdk_span_link in sdk_span.links:
+
+                    collector_span_link = CollectorSpan.Link(
+                        trace_id=(
+                            sdk_span_link.context.trace_id.to_bytes(16, "big")
+                        ),
+                        span_id=(
+                            sdk_span_link.context.span_id.to_bytes(8, "big")
+                        ),
+                    )
+
+                    for key, value in sdk_span_link.attributes.items():
+                        try:
+                            collector_span_link.attributes.append(
+                                AttributeKeyValue(
+                                    **translate_key_values(key, value)
+                                )
+                            )
+                        # pylint: disable=broad-except
+                        except Exception as error:
+                            logger.exception(error)
+
+                    collector_span_kwargs["links"].append(collector_span_link)
+
+            collector_span_kwargs["kind"] = getattr(
+                CollectorSpan.SpanKind, sdk_span.kind.name
+            )
+
+            sdk_resource_instrumentation_library_spans[
+                sdk_span.resource
+            ].spans.append(CollectorSpan(**collector_span_kwargs))
+
+        resource_spans = []
+
+        for (
+            sdk_resource,
+            instrumentation_library_spans,
+        ) in sdk_resource_instrumentation_library_spans.items():
+
+            collector_resource = Resource()
+
+            for key, value in sdk_resource.labels.items():
+
+                try:
+                    collector_resource.attributes.append(
+                        AttributeKeyValue(**translate_key_values(key, value))
+                    )
+                except Exception as error:  # pylint: disable=broad-except
+                    logger.exception(error)
+
+            resource_spans.append(
+                ResourceSpans(
+                    resource=collector_resource,
+                    instrumentation_library_spans=[
+                        instrumentation_library_spans
+                    ],
+                )
+            )
+
+        return ExportTraceServiceRequest(resource_spans=resource_spans)
+
+    def export(self, spans: Sequence[SDKSpan]) -> SpanExportResult:
+        # expo returns a generator that yields delay values which grow
```

Comment:
any advantage of using expo over using backoff's `on_exception` decorator?

Reply:
Yes, sometimes it may be needed to use the delay provided by the server instead of the delay provided by `backoff`, so this approach is preferred.